openapi/src/main/java/io/micronaut/openapi/visitor/ConvertUtils.java

```diff
@@ -188,7 +196,36 @@ public static SecurityRequirement mapToSecurityRequirement(AnnotationValue<io.sw
     }
 
     /**
-     * Detect openapi type nd format by java class name.
+     * Detect openapi type and format for enums.
+     *
+     * @param type enum element
+     * @param schemaType type from swagger Schema annotation
+     * @param schemaFormat format from swagger Schema annotation
+     *
+     * @return pair with openapi type and format
+     */
+    @NonNull
+    public static Pair<String, String> checkEnumJsonValueType(@NonNull EnumElement type, @Nullable String schemaType, @Nullable String schemaFormat) {
+        if (schemaType != null && !schemaType.equals(PrimitiveType.STRING.getCommonName())) {
+            return Pair.of(schemaType, schemaFormat);
+        }
+        Pair<String, String> result = null;
+        // check JsonValue method
+        for (MethodElement method : type.getEnclosedElements(ElementQuery.ALL_METHODS)) {
```

Comment:
this is very expensive, use `getEnclosedElements(ElementQuery.ALL_METHODS).annotated(metadata -> metadata.isAnnotationPresent(JsonValue.class)` to reduce the cost of this